prepackPy/na_counter.py

```diff
@@ -4,20 +4,70 @@
 import warnings
 import pandas as pd
 
-def na_counter(X):
+def na_counter(X, col_index=None):
     """
     summarise the missing data in a dataset
 
     Parameters
     ----------
     X: numpy array, pandas dataframe
         an input dataset
+    col_index: int
 
     Returns
     -------
     na_dict: dictionary
         a summary dictionary (key: value)
         key = an column index of X that has missing values
-        value = an tuple (NA counts, percentage of missing values in each column) 
+        value = an tuple (NA counts, percentage of missing values in each column)
     """
-    return na_dict
\ No newline at end of file
+
+    # input type check
+    if not isinstance(X, (pd.DataFrame, np.ndarray)):
+        raise TypeError("X must be of numpy.ndarray or pandas.DataFrame type")
+
+    if not isinstance(col_index, (int, list, type(None))):
+        raise TypeError("col_index must be of type int or list")
+
+    # input value check
+    if X.shape[0] < 1:
+        raise ValueError("X does not contain any observations")
+
+    if isinstance(col_index, int):
+        if col_index > X.shape[1] - 1 or col_index < 0:
```

Comment:
In python, col_index can be negative. For example, if col_index = -1, it means the last column of X. I guess it would raise an error if `col_index < -X.shape[1]`.

Reply:
Ok, I can incorporate that into func.!

Reply:
Addressed at ` [7c64082](/UBC-MDS/prepackPy/pull/21/commits/7c64082fddfc50c1af566ddcf8433224676c0ab2)`.  Take a look JY, make sure I did it correctly.